client/cli/src/commands/mod.rs

```diff
@@ -162,7 +162,7 @@ macro_rules! substrate_cli_subcommands {
 				}
 			}
 
-			fn base_path(&self) -> $crate::Result<::std::option::Option<::std::path::PathBuf>> {
+			fn base_path(&self) -> $crate::Result<::std::option::Option<::sc_service::config::BasePath>> {
```

Comment:
```suggestion
			fn base_path(&self) -> $crate::Result<::std::option::Option<sc_service::config::BasePath>> {
```

Reply:
Actually I think these are all wrong. sc-cli should probably make a hidden re-export of `sc_service` and the macro should get it from there.

Reply:
Yes that is right.

Reply:
I will make another PR for that for the sake of having isolated commits per change